src/Playroom/Frame/frameMessaging.tsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react';
+
+import { ErrorMessage } from '../RenderError/RenderError';
+
+const PlayroomErrorSource = 'Playroom Frame Error';
+
+export const SendErrorMessage = ({
+  errorMessage,
+}: {
+  errorMessage: string;
+}) => {
+  useEffect(() => {
+    window.parent.postMessage({
+      source: PlayroomErrorSource,
+      message: errorMessage,
+    });
+  }, [errorMessage]);
+
+  return null;
+};
```

Comment:
Could this be a hook? Should it be a hook? E.g. `useRegisterErrorSender`.

Reply:
Its used as a component because its passed to the `Frame` component, and that behaviour is shared across the preview entrypoint too (and that passes more than just a Hook)